### PR TITLE
Create Components

### PR DESCRIPTION
**Pull request by mnekx**
## Description

In this milestone, foundations are laid for your Bookstore website. Creation of the React part of the app.  Also, Routing using React Router is set up.

## Type of change
The following are the changes done

- [x] Initialize React app.
- [x] Create a directory for components.
- [x] Add [React Router](https://reactrouter.com/web/guides/quick-start) and set two <Route>s and <Link>s for the app's navigation:
- [x] Display the list of books (empty at this point but it should be ready for the data) with the Remove button (no functionality yet).
- [x] Create a component called Book for displaying a single book (with a title and an author) and reuse it in a component that displays a list of books.
- [x] Should have a form for adding a book (no functionality yet).
- [x] Create a separate component for this form (with inputs for a title and an author).
- [x] Categories:
        Should display a button "Check status" only.

## Checklist:

- [x] No linter errors.
- [x] I have used correct Gitflow.
- [x] Work is documented in a professional way.
- [x] List of JavaScript best practices is followed.

### CODE REVIEW

File: src/components/AddBook.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+      <legend>Add new Book</legend>
+      <label htmlFor="title">
+        <input type="text" id="title" name="title" placeholder="Book Title" />
+      </label>
+      <label htmlFor="author">
+        <input type="text" id="author" name="author" placeholder="Author" />
+      </label>
+      <button type="submit">Add Book</button>
+    </fieldset>
+  </form>
```
Comment (dasileker) [inline src/components/AddBook.js:15]: - [x] Hello @mnekx, as you are in react module you should know that's not how react works, react components are not just a simple form please read this [Article](https://www.digitalocean.com/community/tutorials/how-to-create-custom-components-in-react)

**Review by dasileker — CHANGES_REQUESTED** [review]
### **Hi @mnekx,** 

### **Good job so far!**
There are some issues that you still need to work on to go to the next project but you are almost there!

## **Highlights**
- Nice PR 👌.
- Good Readme file 👌.
- Linters are passing 👌.


## Required Changes ♻️
_Check the comments under the review._

- Nice Job on the bookstore app however the components you have created are not correctly coded, please refer to [This Article](https://reactjs.org/docs/react-component.html?msclkid=05040d1aab6511ec8f36e80a7d00821f) 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/AddBook.js (hunk 1 above)
Comment (mnekx) [reply src/components/AddBook.js:15]: I have read the [article](https://www.digitalocean.com/community/tutorials/how-to-create-custom-components-in-react) but I think  I am missing your point 😞, I would appreciate if you were kind enough to show me what exactly is wrong with the components.

Comment (mnekx) [reply src/components/AddBook.js:15]: In the code I am trying to use functional components like so in the [article](https://www.digitalocean.com/community/tutorials/how-to-create-custom-components-in-react):
image. Please help me figure this out 🤔

Comment (dasileker) [reply src/components/AddBook.js:15]: - [x] hello @mnekx, maybe I didn't explain it clearly you are not using `import React from'react'`.

Comment (mnekx) [reply src/components/AddBook.js:15]: Thanks

**Review by mikethreels — APPROVED** [review]
Hi Chambuya, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

### Please avoid requesting a review before you made the required changes.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**